python/do.py

```diff
@@ -0,0 +1,199 @@
+        params: If the above queries are parameterized, provide a {key: value} map of parameters applied to the given queries.
+
+    Returns:
+        value: {field_name: field_value} map containing the result records of the evaluated query.
+    """
+
+    if params is None:
+        params = {}
+
+    conditions = conditionals[::2]
+    if_queries = conditionals[1::2]
+
+    if len(conditions) != len(if_queries):
+        raise ValueError(WRONG_STRUCTURE_MSG)
+
+    if any(not isinstance(condition, bool) for condition in conditions) or any(
+        not isinstance(if_query, str) for if_query in if_queries
+    ):
+        raise ValueError(WRONG_STRUCTURE_MSG)
+
+    memgraph = gqlalchemy.Memgraph()
+
+    for condition, if_query in zip(conditions, if_queries):
+        if condition:
+            results = _execute_and_fetch_parametrized(
+                memgraph, if_query, parameters=params
+            )
+
+            return [
+                mgp.Record(
+                    value={
+                        field_name: _gqlalchemy_type_to_mgp(ctx.graph, field_value)
+                        for field_name, field_value in result.items()
+                    }
+                )
+                for result in results
+            ]
+
+    results = _execute_and_fetch_parametrized(memgraph, else_query, parameters=params)
+
+    return [
+        mgp.Record(
+            value={
+                field_name: _gqlalchemy_type_to_mgp(ctx.graph, field_value)
+                for field_name, field_value in result.items()
+            }
+        )
+        for result in results
+    ]
+
+
+@mgp.read_proc
+def when(
+    ctx: mgp.ProcCtx,
+    condition: bool,
+    if_query: str,
+    else_query: mgp.Nullable[str] = "",
+    params: mgp.Nullable[mgp.Map] = None,
+) -> mgp.Record(value=mgp.Nullable[mgp.Map]):
+    """Depending on the value of the condition, executes `if_query` or `else_query` (read-only).
+
+    Args:
+        condition: The boolean value that determines what query to execute.
+        if_query: The read-only query to be executed if the condition is satisfied.
+        else_query: The read-only query to be executed if the condition is not satisfied.
+        params: {key: value} map of parameters applied to the given queries.
+
+    Returns:
+        value: {field_name: field_value} map containing the result records of the evaluated query.
+    """
+
+    if params is None:
+        params = {}
+
+    memgraph = gqlalchemy.Memgraph()
+
+    results = _execute_and_fetch_parametrized(
+        memgraph, if_query if condition else else_query, parameters=params
+    )
+
+    return [
+        mgp.Record(
+            value={
+                field_name: _gqlalchemy_type_to_mgp(ctx.graph, field_value)
+                for field_name, field_value in result.items()
+            }
+        )
+        for result in results
+    ]
+
+
+def _get_edge_by_id(
```

Comment:
Maybe a bit better naming like get edge from starting node or something similar because my first impression when I see such name is that everything will be searched.

Reply:
Done